src/redux/selectors/benefits/benefitsDataSelectors.js

```diff
@@ -72,14 +74,43 @@ export const eligibleBenefitsSelector = createSelector(
   benefitsEligibilitySelector,
   (lang, benefitsData, benefitsDataFr, benefitsEligibility) => {
     if (!benefitsEligibility) return [];
+    var internalBenefits = [];
     if (lang === "fr") {
-      return benefitsEligibility.map((value) =>
-        benefitTransformer(benefitsDataFr[value])
-      );
+      benefitsEligibility.forEach(function (item) {
+        if (benefitsData[item]["service_type"] === "Internal")
+          internalBenefits.push(benefitTransformer(benefitsDataFr[item]));
+      });
+      return internalBenefits;
+    } else {
+      benefitsEligibility.forEach(function (item) {
+        if (benefitsData[item]["service_type"] === "Internal")
+          internalBenefits.push(benefitTransformer(benefitsData[item]));
+      });
+      return internalBenefits;
+    }
+  }
+);
+
+export const externalBenefitsDataSelector = createSelector(
+  languageSelector,
+  benefitsMapSelector,
+  benefitsMapSelectorFr,
+  benefitsEligibilitySelector,
+  (lang, benefitsData, benefitsDataFr, eligibleBenefits) => {
+    if (!eligibleBenefits) return [];
+    var externalBenefits = [];
```

Comment:
lots of unnecessary code, can be solved by using a map function, see other selectors